src/dbus/connection.c

```diff
@@ -259,6 +260,18 @@ int connection_dequeue(Connection *connection, Message **messagep) {
                 return error_fold(r);
         }
 
+        if (fdlist_count(message->fds) > 0) {
```

Comment:
`message` might be NULL. This shouldn't pass the test suite.

Reply:
Indeed. Fixed and force-pushed. Still issues outstanding.